tests/datasource/fluent/test_pandas_datasource.py

```diff
@@ -486,6 +493,8 @@ def test_cloud_get_csv_asset_not_in_memory(valid_file_path: pathlib.Path):
 
     assert csv_asset_name not in context.data_sources.all()._in_memory_data_assets
 
+    context.data_sources.delete(name=datasource_name)
```

Comment:
If this is meant to clean up we should do it in a `try/finally` block. For example, if the assert fails in the line above, this will never be run.